### ISSUE TITLE
ReactRouterRelay.Container to pass down props?

### ISSUE THREAD

**Issue by sgwilym** [relay-tools/react-router-relay#18]
For application state it's sometimes useful to clone `children` with props like a flux or redux store. However, this just means I'm putting those props on the `ReactRouterRelay.Container` — would it be possible for it to pass those props on?

**Comment by taion** [comment]
How are you attaching those extra props?

**Comment by sgwilym** [comment]
The way I used to do it before using this package was my own createElement function with a HOC like this (in this case, with Flummox's FluxComponent):

```jsx
const flux = new flux();

function createElement(Component, props) {
  return <FluxComponent flux={flux} connectToStores={['application']}>
    <Component {...props}/>
  </FluxComponent>;
}
```

As you know, react-router-relay uses createElement so that's out.

But it's not just restricted to that situation: let's say I have a component that would want to pass down a callback function to its children:

```jsx
render() {
  const beeper = this.beep;
  
  const childrenWithProps = React.Children.map(this.props.children, (el) => {
    return React.cloneElement(el, {beep: beeper})
  })

  return(
    <div className={styles.root}>
      { childrenWithProps }
    </div>
  );
}
```

The component represent by a nested route will never get the `beep` prop — `ReactRouterRelay.Container` will — and now I can't beep anymore.

**Comment by taion** [comment]
Why wouldn't you handle this as e.g.:

```js
function myCreateElement(Component, props) {
  return ReactRouterRelay.createElement(Component, {beep, ...props});
}
```

**Comment by sgwilym** [comment]
Duh. Thanks, I was starting to get the feeling I might be missing something obvious!

**Comment by taion** [comment]
Oh, this does break the `cloneElement` on parent to child. Okay - I wanted to use the separate `routerProps` thing to avoid potential name conflicts with the `Component` prop (which the `Container` also needs), but I guess that's not safe for this use case.

Care to pull together a PR?

**Comment by sgwilym** [comment]
Even though your last suggestion fixed my immediate problem, I could give it a shot :)

**Comment by taion** [comment]
I think it'd basically be using `{...props}` here: https://github.com/relay-tools/react-router-relay/blob/v0.4.1/src/createElement.js#L9 and `...routerProps` here: https://github.com/relay-tools/react-router-relay/blob/v0.4.1/src/Container.js#L26.

**Comment by taion** [comment]
Fixed by #20.
